assets/src/blocks/TopicLink/TopicLinkEditor.js

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable no-shadow */
+
+const {useSelect} = wp.data;
+const {
+  BlockControls,
+  MediaUpload,
+  MediaUploadCheck,
+  InspectorControls,
+} = wp.blockEditor;
+const {
+  SelectControl,
+  PanelBody,
+  ToolbarGroup,
+  ToolbarButton,
+  FocalPointPicker,
+} = wp.components;
+const {sprintf, __} = wp.i18n;
+
+/**
+ * TopicLinkEditor component for managing the Topic Link block in the editor.
+ *
+ * @param {Object} props - Component properties.
+ * @param {Object} props.attributes - Block attributes.
+ * @param {boolean} props.isSelected - Indicates if the block is selected.
+ * @param {Function} props.setAttributes - Function to update block attributes.
+ * @return {JSX.Element} The Topic Link Editor component.
+ */
+export const TopicLinkEditor = ({
+  attributes,
+  isSelected,
+  setAttributes,
+}) => {
+  const {
+    categoryId,
+    categoryName,
+    focal_points,
+    imageId: customImageId,
+    imageUrl: customImageFromId,
+  } = attributes;
+
+  const {
+    categoriesList,
+    imageId,
+    imageUrl,
+    imageAlt,
+    currentPostCategories,
+  } = useSelect(select => {
+    const categoriesList = [].concat(
+      select('core').getEntityRecords('taxonomy', 'category', {
+        hide_empty: true,
+        per_page: -1,
+      }) || []
```

Comment:
Not mandatory but missing `;` at EOL

Reply:
I think it's not necessary:

<hr>

Using ";"
![Screenshot from 2024-12-18 10-03-25](https://github.com/user-attachments/assets/cab314ee-b45e-455e-a116-a01edc8bfdd7)

<hr>

Using ","
![Screenshot from 2024-12-18 10-03-33](https://github.com/user-attachments/assets/70d0c944-d1e3-461c-aea2-5e26411ef6fc)

Reply:
Oh you're correct! I've looked wrongly 🙏